src/framework/scene-registry.js

```diff
@@ -253,6 +261,38 @@ class SceneRegistry {
         }
     }
 
+    _addHierarchyToScene(url, data, callback) {
+        const self = this;
+
+        // Because we need to load scripts before we instance the hierarchy (i.e. before we create script components)
+        // Split loading into load and open
+        const handler = this._app.loader.getHandler("hierarchy");
+
+        // called after scripts are preloaded
+        const _loaded = function () {
+            self._app.systems.script.preloading = true;
+            const entity = handler.open(url, data);
+
+            self._app.systems.script.preloading = false;
+
+            // clear from cache because this data is modified by entity operations (e.g. destroy)
+            self._app.loader.clearCache(url, "hierarchy");
+
+            // add to hierarchy
+            self._app.root.addChild(entity);
+
+            // initialize components
+            self._app.systems.fire('initialize', entity);
+            self._app.systems.fire('postInitialize', entity);
+            self._app.systems.fire('postPostInitialize', entity);
+
+            if (callback) callback(null, entity);
+        };
+
+        // load priority and referenced scripts before opening scene
+        self._app._preloadScripts(data, _loaded);
+    }
```

Comment:
```suggestion
    _addHierarchyToScene(url, data, callback) {
        const app = this._app;

        // Because we need to load scripts before we instance the hierarchy (i.e. before we create script components)
        // Split loading into load and open
        const handler = app.loader.getHandler("hierarchy");

        // called after scripts are preloaded
        const _loaded = function () {
            app.systems.script.preloading = true;
            const entity = handler.open(url, data);

            app.systems.script.preloading = false;

            // clear from cache because this data is modified by entity operations (e.g. destroy)
            app.loader.clearCache(url, "hierarchy");

            // add to hierarchy
            app.root.addChild(entity);

            // initialize components
            app.systems.fire('initialize', entity);
            app.systems.fire('postInitialize', entity);
            app.systems.fire('postPostInitialize', entity);

            if (callback) callback(null, entity);
        };

        // load priority and referenced scripts before opening scene
        app._preloadScripts(data, _loaded);
    }
```

Reply:
Is that property lookup that expensive? Happy to change it but interested in the performance difference.

Reply:
I didn't benchmark 😅 Probably good to reduce code-size/ascii-soup aswell

Reply:
Done